### PR TITLE
Added documentation Local driver database files

### PR DESCRIPTION
**Pull request by chensixx**
Added documentation for functions in localDocumentDealtaConnection, localSessionStorageDb and the ICollection interface.

### CODE REVIEW

File: server/routerlicious/packages/services-core/src/database.ts [hunk 1]
```diff
@@ -33,20 +33,73 @@ export interface IDatabaseManager {
...
     findOrCreate(query: any, value: T): Promise<{ value: T, existing: boolean }>;
 
+    /**
+     *
+     * Finds the query in the db. If it exists, update the value to set.
+     * Throws if query cannot be found.
+     *
+     *  @param filter - data we want to find
+     *  @param set - new values to change to
+     *  @param addToSet
```
Comment (ChumpChief) [inline server/routerlicious/packages/services-core/src/database.ts:73]: What does this param do?

**Review by ChumpChief — APPROVED** [review]
Would be good to clarify the "any" types a bit more (particularly things like filter/query/sort).  As you're making updates please also make sure to use consistent whitespace and terminology (e.g. "database" vs. "db").

File: server/routerlicious/packages/services-core/src/database.ts (hunk 1 above)
Comment (chensixx) [reply server/routerlicious/packages/services-core/src/database.ts:73]: i dont know...it's not used anywhere except in the mongodb.ts, where it's also ambiguous what it's used for

**Comment by tylerbutler** [comment]
@chensixx Is this PR still relevant?

**Comment by chensixx** [comment]
let me double check

File: packages/drivers/local-driver/src/localDocumentDeltaConnection.ts [hunk 2]
```diff
@@ -16,9 +16,23 @@ import { LocalWebSocketServer } from "@fluidframework/server-local-server";
...
+ */
 export class LocalDocumentDeltaConnection
     extends DocumentDeltaConnection
     implements IDocumentDeltaConnection {
+    /**
+     * Create a LocalDocumentDeltaConnection
+     * Handle initial messages, contents or signals if they were in queue
+     *
+     * @param tenantId - the ID of the tenant
+     * @param id - document Id
```
Comment (ChumpChief) [inline packages/drivers/local-driver/src/localDocumentDeltaConnection.ts:31]:
```suggestion
     * @param id - document ID
```

File: packages/drivers/local-driver/src/localSessionStorageDb.ts [hunk 3]
```diff
@@ -8,15 +8,23 @@ import { ITestDbFactory } from "@fluidframework/server-test-utils";
...
     private readonly collectionName: string;
     constructor(namespace, name) {
         this.collectionName = `${namespace}-${name}`;
     }
 
+    /**
+     * {@inheritDoc @fluidframework/server-services-core#ICollection.find}
+     *
+     *  Each query key consists of several keys separated by '.' e.g: "operation.sequenceNumber".
+     *  The hierarchical syntax allows finding nested key patterns.
```
Comment (ChumpChief) [inline packages/drivers/local-driver/src/localSessionStorageDb.ts:24]:
```suggestion
     * Each query key consists of several keys separated by '.' e.g: "operation.sequenceNumber".
     * The hierarchical syntax allows finding nested key patterns.
```

File: packages/drivers/local-driver/src/localSessionStorageDb.ts [hunk 4]
```diff
@@ -150,6 +204,13 @@ class LocalSessionStorageCollection<T> implements ICollection<T> {
         return values;
     }
 
+    /**
+     * Inserts values into the session storge.
+     * Values are expected to have a member "_id" which is a unique id, otherwise will be assigned one
+     *
+     * @param values - data to insert to the database
+     *
+     */
```
Comment (ChumpChief) [inline packages/drivers/local-driver/src/localSessionStorageDb.ts:213]:
```suggestion
     * @param values - data to insert to the database
     */
```

File: packages/drivers/local-driver/src/localSessionStorageDb.ts [hunk 5]
```diff
@@ -161,15 +222,25 @@ class LocalSessionStorageCollection<T> implements ICollection<T> {
...
     }
 
+    /**
+     * Finds the query in session storage and returns its value.
+     * Returns null if query is not found.
+     * Query is expected to have a member "_id" which is a unique id.
+     *
+     * @param query - what to find in the database
+     *
+    */
```
Comment (ChumpChief) [inline packages/drivers/local-driver/src/localSessionStorageDb.ts:232]:
```suggestion
     * @param query - what to find in the database
     */
```

File: packages/drivers/local-driver/src/localSessionStorageDb.ts [hunk 6]
```diff
@@ -161,15 +222,25 @@ class LocalSessionStorageCollection<T> implements ICollection<T> {
...
     private findOneInternal(query: any): any {
         if (query._id) {
-            const json = sessionStorage.getItem(`${this.collectionName}-${query._id}`);
+            const json = sessionStorage.getItem(`${this.collectionName}-${query._id}`);// form of data
```
Comment (ChumpChief) [inline packages/drivers/local-driver/src/localSessionStorageDb.ts:235]:
```suggestion
            const json = sessionStorage.getItem(`${this.collectionName}-${query._id}`);
```

File: packages/drivers/local-driver/src/localSessionStorageDb.ts [hunk 7]
```diff
@@ -161,15 +222,25 @@ class LocalSessionStorageCollection<T> implements ICollection<T> {
...
             if (json) {
                 // eslint-disable-next-line @typescript-eslint/no-unsafe-return
                 return JSON.parse(json);
             }
         } else {
             const queryKeys = Object.keys(query);
+
+            // Optimization needed
```
Comment (ChumpChief) [inline packages/drivers/local-driver/src/localSessionStorageDb.ts:243]:
```suggestion
```

Either create an issue and reference the issue number here or remove comment (currently unclear what work is needed)

File: server/routerlicious/packages/services-core/src/database.ts [hunk 8]
```diff
@@ -32,21 +32,82 @@ export interface IDatabaseManager {
     getScribeDeltaCollection(tenantId: string, documentId: string): Promise<ICollection<ISequencedOperationMessage>>;
 }
 
+/**
+ * Interface for a database of values that have type T.
+ * In some implementations,T should have a member "_id" which is a string used
```
Comment (ChumpChief) [inline server/routerlicious/packages/services-core/src/database.ts:37]:
```suggestion
 * In some implementations, T should have a member "_id" which is a string used
```

File: server/routerlicious/packages/services-core/src/database.ts [hunk 9]
```diff
@@ -32,21 +32,82 @@ export interface IDatabaseManager {
...
+     * Insert the value if query was not found.
+     *
+     * @param query - data we want to find
+     * @param value - data to insert to the database if we cannot find query
+     */
     findOrCreate(query: any, value: T): Promise<{ value: T, existing: boolean }>;
 
+    /**
+     *
+     * Finds the query in the database. If it exists, update the value to set.
```
Comment (ChumpChief) [inline server/routerlicious/packages/services-core/src/database.ts:74]:
```suggestion
    /**
     * Finds the query in the database. If it exists, update the value to set.
```

File: server/routerlicious/packages/services-core/src/database.ts [hunk 10]
```diff
@@ -32,21 +32,82 @@ export interface IDatabaseManager {
...
+     * Throws if query cannot be found.
+     *
+     *  @param filter - data we want to find
+     *  @param set - new values to change to
+     *  @param addToSet - an operator that adds a value to the database unless the value already exists;
+     *   only used in mongodb.ts
```
Comment (ChumpChief) [inline server/routerlicious/packages/services-core/src/database.ts:80]:
```suggestion
     * @param filter - data we want to find
     * @param set - new values to change to
     * @param addToSet - an operator that adds a value to the database unless the value already exists;
     * only used in mongodb.ts
```

File: server/routerlicious/packages/services-core/src/database.ts [hunk 11]
```diff
@@ -32,21 +32,82 @@ export interface IDatabaseManager {
...
+     */
     update(filter: any, set: any, addToSet: any): Promise<void>;
 
+    /**
+     *
+     * Finds the value that satisfies query. If it exists, update the value to new set.
```
Comment (ChumpChief) [inline server/routerlicious/packages/services-core/src/database.ts:86]:
```suggestion
    /**
     * Finds the value that satisfies query. If it exists, update the value to new set.
```

File: server/routerlicious/packages/services-core/src/database.ts [hunk 12]
```diff
@@ -32,21 +32,82 @@ export interface IDatabaseManager {
...
+     * Otherwise inserts the set to the datbase.
+     *
+     *  @param filter - data we want to find
+     *  @param set - new values to change to
+     *  @param addToSet - an operator that adds a value to the database unless the value already exists;
+     *   only used in mongodb.ts
```
Comment (ChumpChief) [inline server/routerlicious/packages/services-core/src/database.ts:92]:
```suggestion
     * @param filter - data we want to find
     * @param set - new values to change to
     * @param addToSet - an operator that adds a value to the database unless the value already exists;
     * only used in mongodb.ts
```

File: server/routerlicious/packages/services-core/src/database.ts [hunk 13]
```diff
@@ -32,21 +32,82 @@ export interface IDatabaseManager {
...
+     */
     upsert(filter: any, set: any, addToSet: any): Promise<void>;
 
+    /**
+     * Inserts an entry into the database.
+     * Throws if it would overwrite an existing entry
+     *
+     *  @param value - data to insert to the database
+     *
+     */
```
Comment (ChumpChief) [inline server/routerlicious/packages/services-core/src/database.ts:102]:
```suggestion
     * @param value - data to insert to the database
     */
```

**Review by ChumpChief — APPROVED** [review]
Just some nits, looks good

**Pull request merged**